kratos/utilities/function_parser_utility.cpp

```diff
@@ -0,0 +1,183 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//  Collaborator:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+#include "tinyexpr/tinyexpr/tinyexpr.h"
+
+// Project includes
+#include "utilities/function_parser_utility.h"
+#include "utilities/string_utilities.h"
+
+namespace Kratos
+{
+
+GenericFunctionUtility::GenericFunctionUtility(
+    const std::string& rFunctionBody,
+    Parameters LocalSystem
+    ) : mOriginalFunctionBody(rFunctionBody),
+        mFunctionBody(rFunctionBody)
+{
+    // Correcting function from python style to tinyexpr
+    std::unordered_map<std::string,std::string> aux_replace_letters({{"**","^"},{"X","a"},{"Y","b"},{"Z","c"}});
```

Comment:
why is this necessary?

Reply:
Because tinyexpr does not work with capital letters (it is in the documentation BTW)

Reply:
ok

Reply:
can you add a comment?
Also why you replace e.g. `X` with `a` would make it easier to understand

Reply:
i would do X->x0 Y->y0 etc